### ISSUE TITLE
Date selection and navigation not working on Mac OS X 10.12.4 under Java 8 update 121

### ISSUE THREAD

**Issue by raffi-minassian** [LGoodDatePicker/LGoodDatePicker#49]
As of Java 8 update 121, clicking to select a date no longer works. Also, when navigating between months using the forward and back buttons, the display redraw errors causing dates to misalign in the grid.

I've attached a gif showing the issue using the Demo jar provided from the project on github project.

```
java -version
java version "1.8.0_121"
Java(TM) SE Runtime Environment (build 1.8.0_121-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.121-b13, mixed mode)
```

The problem does NOT occur under java8 update 111.

![lgooddate-issue](https://cloud.githubusercontent.com/assets/2666097/25094653/32b61d1e-2366-11e7-896e-fef1fd501761.gif)

**Comment by BlakeTNC** [comment]
Hello @raffi-minassian , 
I installed Java 8 update 121 on my Windows 8.1 machine, and could not reproduce this bug. That appears to narrow down the source of this problem. It looks like it does not occur on -all- installations of Java update 121, and it's possible that it only occurs on update 121 on -Mac OS X- (or possibly also other Linux-based operating systems). 

I don't own any Mac OS X machines, so compiling and running on update 121 was about as far as I could go with troubleshooting this particular issue. ("I can't reproduce.") However, my inability to reproduce it does not mean there is no problem. Do you have the programming skills to debug this issue on your Mac OS X machine? (Or know someone who does?)

Warm regards,
Blake

**Comment by raffi-minassian** [comment]
Hi @BlakeTNC . Thanks for your reply. I will fork the project and see if I can find some time next week to find and fix the issue. Then I will submit a pull request and see what you think.

Thanks!

**Comment by BlakeTNC** [comment]
Update: 
Another person has reported this same problem, which further confirms its existence on the Macintosh. I would like to post a copy of my reply to them, just so that it is clear to anyone reading this that my hands are a tied in fixing this particular bug.

This is the situation for me:
"Unfortunately this problem does not occur on Windows, and I do not own a Macintosh. Therefore, I personally have no way of fixing the problem (except by supporting the efforts of other developers who choose to work on it). If this particular problem is going to be debugged and fixed, it's going to have to be done by a programmer who owns a Mac. Any programmer with the prerequisite skills can send a pull request on GitHub for LGoodDatePicker if they develop a fix. I feel kind of bad about this situation, because I have promptly fixed every other major problem that ever occurred with the date picker. (Although there were not many problems, most new developments were feature requests.) I would hate to see a major problem go unresolved. -Blake"

**Comment by BlakeTNC** [comment]
Hello everyone,
This bug has been fixed, and a new release was made which includes the fix. The new release is [LGoodDatePicker 10.1.2](https://github.com/LGoodDatePicker/LGoodDatePicker/releases).

Many thanks to André, who provided me with remote access to a MacOSX instance so that I could investigate and fix the problem. 

Note: I also plan to release this to Maven Central, and as a LGoodDatePicker backport, but I intend to wait  a week or two until I can get some user feedback on the fix. If you were affected by this bug, please let me know if the new release is working properly for you.

For anyone curious about the technical details:
There may or may not be a bug in Java 121 or higher, on MacOSX. The date picker popup window was sometimes opening behind the parent window, and sometimes opening on top of the parent window, but the click behavior was broken regardless of where the popup window was drawn. (Programmatically, the parent window may have been blocking some or all of the clicks.) My fix was to set the CustomPopup class "displayWindow" component to be "always on top". This fixed the bug, and this setting seems appropriate enough for a date picker.

warm regards,
Blake

**Comment by raffi-minassian** [comment]
Hello all. I'm confirming that I'm seeing this problem as resolved testing on Mac OS 10.12.4 with Java 1.8.0_131.

**Comment by BlakeTNC** [comment]
Hello All,
[LGoodDatePicker 10.2.3](https://github.com/LGoodDatePicker/LGoodDatePicker/releases) (with the fix for the MacOSX/Java 121 Issue), was added to [Maven Central](http://search.maven.org/#search%7Cga%7C1%7Ca%3A%22LGoodDatePicker%22), and to the [LGoodDatePicker Backports](https://github.com/LGoodDatePicker/LGoodDatePicker/releases/tag/Backports). 
(It might take up to a day to appear on Maven Central.)
Warm regards,
Blake
